plugin/percona-pam-for-mysql/src/groups.c

```diff
@@ -17,19 +17,21 @@ Foundation, Inc., 59 Temple Place, Suite 330, Boston, MA 02111-1307 USA
 
 #include <pwd.h>
 #include <grp.h>
+#include <errno.h>
+#include <my_sys.h>
 #include <string.h>
 #include <stdlib.h>
 #include <unistd.h>
 
-enum { max_nss_name_len = 10240 };
-enum { max_number_of_groups = 1024 };
+static int32_t gr_buf_size= 0;
```

Comment:
"size_t" is unsigned, sysconf returns "long" which is signed. And I wanted to avoid mutexes and still feel safe to load/store gr_buf_size from multiple threads. Since on x86 32-bit load/store instructions are atomic, I felt like int32 would be OK.